### PR TITLE
Cleanup Onboarding components (wave 1/n)

### PR DESCRIPTION
**Pull request by SamChou19815**
### Summary 

The current onboarding components are in a mess. Many code are untyped or with `@ts-ignore`...

This diff cleans them up with the following major changes:

1. Collapsed some different css names with exactly the same style into a single class, so that it becomes very clear that some code can be reused (plan for next diff)
2. Removed some unused code in OnboardingBasic.
3. Remove all the id property in vue templates: they are never used...
4. OnboardingBasic.vue is now in TS!
5. Fix all the type errors in OnboardingBasic.

Before this change, the data of OnboardingBasic is a mess. In order to accomodate the incoming user.majors and user.minors being arrays, the code added a lot of if-else condition here and there to deal with it. It leaves us with some code that is almost impossible to statically type. I changed it so that the data is flattened when the component mounts instead of when every time data changes, so we no longer need to deal with two different data format everywhere. As a result, the code is much cleaner and the data is in a consistent format that's easier to write type definitions.

Linter warnings: 47 -> 44

### Test Plan 

Play with the onboarding component to ensure all the data are still saved and there are no visual regressions

### Notes 

After this PR, it becomes clear that we can extract out a common dropdown component for onboarding instead of passing the big display options. I will address it in a future PR.

### CODE REVIEW

File: src/components/Modals/Onboarding/OnboardingBasic.vue [hunk 1]
```diff
@@ -522,43 +413,41 @@ export default Vue.extend({
...
       this.closeDropdownIfOpen('college', i);
     },
-    closeMajorDropdownIfOpen(event, i) {
+    closeMajorDropdownIfOpen(_: unknown, i: number) {
       this.closeDropdownIfOpen('major', i);
     },
-    closeMinorDropdownIfOpen(event, i) {
+    closeMinorDropdownIfOpen(_: unknown, i: number) {
       this.closeDropdownIfOpen('minor', i);
     },
```
Comment (tcho6319) [inline src/components/Modals/Onboarding/OnboardingBasic.vue:424]: I feel like these functions are repetitive and could be refactored into one function. Could this be handled when factoring out a dropdown component?

Comment (tcho6319) [reply src/components/Modals/Onboarding/OnboardingBasic.vue:424]: Similarly for the functions `selectCollege()`, `selectMajor()`, `selectMinor()` and `removeCollege()`, etc.

File: src/components/Modals/Onboarding/OnboardingBasic.vue [hunk 2]
```diff
@@ -461,35 +380,8 @@ export default Vue.extend({
...
-            acronym: option.acronym,
-            fullName: option.placeholder,
-          };
-
-          list.push(obj);
-        }
-      });
-
-      return list;
-    },
```
Comment (tcho6319) [inline src/components/Modals/Onboarding/OnboardingBasic.vue:489]: Just checking but is this dead code?

**Review by tcho6319 — COMMENTED** [review]
LGTM Sam! Left a couple quick clarification questions and then should be good for approval. Thank you so much for doing this.

File: src/components/Modals/Onboarding/OnboardingBasic.vue (hunk 2 above)
Comment (SamChou19815) [reply src/components/Modals/Onboarding/OnboardingBasic.vue:489]: Yes this is dead. The vue template isn't using it

File: src/components/Modals/Onboarding/OnboardingBasic.vue (hunk 1 above)
Comment (SamChou19815) [reply src/components/Modals/Onboarding/OnboardingBasic.vue:424]: Yes all of them will be handled

**Review by tcho6319 — APPROVED** [review]
LGTM!

**Pull request merged**